src/display/process/allProcs.go

```diff
@@ -268,12 +279,44 @@ func AllProcVisuals(dataChannel chan []*proc.Process,
 							killSelected = false
 							myPage.ProcTable.CursorColor = selectedStyle
 						}
+						sendSignal = false
+						updateUI()
 					case "K", "<F9>":
 						// get process and kill it
 						procToKill, err := proc.NewProcess(pidToKill)
 						myPage.ProcTable.CursorColor = selectedStyle
 						if err == nil {
-							err = procToKill.Kill()
+							err = procToKill.SendSignal(syscall.SIGTERM)
+							if err != nil {
+								myPage.ProcTable.CursorColor = errorStyle
+							}
+						} else {
+							myPage.ProcTable.CursorColor = errorStyle
+						}
+						runAllProc = true
+						killSelected = false
+						updateProcs()
+					case "j", "<Down>":
+						signals.Table.ScrollDown()
+						ui.Render(signals)
+					case "k", "<Up>":
+						signals.Table.ScrollUp()
+						ui.Render(signals)
+					case "1", "2", "3", "4", "5", "6", "7", "8", "9":
+						scrollIdx, _ := strconv.Atoi(e.ID)
+						if utils.Contains([]string{"1", "2", "3"}, previousKey) != -1 {
+							prevIdx, _ := strconv.Atoi(previousKey)
+							scrollIdx = 10*prevIdx + scrollIdx
+							previousKey = "g" // TODO: don't use this hack
```

Comment:
Maybe just set it to an empty string here?

Reply:
That doesn't work, because there is a check at the end of the switch statement that sets it back to e.ID. So we need to refactor to set previousKey only when we know we will check that key later, inside the relevant cases instead of doing it together at the end

Reply:
I'm a little confused, but as long as you've got an approach in mind that you think works, should be fine, do try it out.

Reply:
Can you check if the changes I've made to this now make sense?